py/src/braintrust/logger.py

```diff
@@ -1372,6 +1421,58 @@ def __str__(self):
         )
 
 
+class ExperimentDataset:
```

Comment:
Would be cool to have a base class (i.e. an interface) for this and `Dataset` to implement. I imagine we'd have a matching interface on the JS side. Similar to the `Span` abstract base class.